### PR TITLE
apiTS is made long, queue length added to bridge message queue, stats

### PR DESCRIPTION
**Pull request by karthikbr82**
format changed for sending it to backend, handling keepalive thread in a
better way.

@marcinag @eljasala @sli-cs  PTAL

### CODE REVIEW

File: callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsBridgeKeepAliveManager.java [hunk 1]
```diff
@@ -123,6 +123,13 @@ public void shutDownKeepAliveSender() {
 	public void startKeepAliveSender(String authToken) {
 		logger.info("Starting keepAlive Sender");
 		this.token = authToken;
+		
+		//make sure there is no thread running already
+		scheduler.shutdownNow();
```
Comment (sli-cs) [inline callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsBridgeKeepAliveManager.java]: if we shut it down here, then I guess the next line call:
```scheduler.scheduleAtFixedRate()``` might throw exception.

**Comment by marcinag** [comment]
Pivotal story?

File: callstats-java-sdk/src/main/java/io/callstats/sdk/internal/BridgeStatusInfoQueue.java [hunk 2]
```diff
@@ -18,6 +19,10 @@ public int getLength() {
 	}
 
 	public void push(BridgeStatusInfo bridgeStatusInfo) {
+		if (getLength() > MAX_QUEUE_LENGTH) {
```
Comment (sli-cs) [inline callstats-java-sdk/src/main/java/io/callstats/sdk/internal/BridgeStatusInfoQueue.java]: shall we add a 'warn' log here?

File: callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsBridgeKeepAliveManager.java [hunk 3]
```diff
@@ -121,8 +123,10 @@ public void shutDownKeepAliveSender() {
 	 * Start keep alive sender.
 	 */
 	public void startKeepAliveSender(String authToken) {
+		this.token = authToken;
```
Comment (eljasala) [inline callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsBridgeKeepAliveManager.java]: What happens if this is executed by multiple threads at same time?

Comment (karthikbr82) [inline callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsBridgeKeepAliveManager.java]: u mean multiple intialize in mutiple threads?

Comment (eljasala) [inline callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsBridgeKeepAliveManager.java]: Yeah

File: callstats-java-sdk/src/main/java/io/callstats/sdk/CallStats.java [hunk 4]
```diff
@@ -418,6 +403,25 @@ public synchronized void reportConferenceStats(String userID, ConferenceStats st
...
 	
+	private synchronized void startKeepAliveThread() {
+		if (bridgeKeepAliveManager == null) {
+			bridgeKeepAliveManager = new CallStatsBridgeKeepAliveManager(appId, bridgeId, authenticator.getToken(), httpClient, new CallStatsBridgeKeepAliveStatusListener() {
+				public void onKeepAliveError(
+						CallStatsErrors error,
+						String errMsg) {
+					if (error == CallStatsErrors.AUTH_ERROR) {
+						authenticator.doAuthentication();
+					}
```
Comment (marcinag) [inline callstats-java-sdk/src/main/java/io/callstats/sdk/CallStats.java]: Error should be logged.

**Comment by eljasala** [comment]
LGTM

**Pull request merged**